src/heitech.zer0mqXt.core/ISocket.cs

```diff
@@ -5,20 +5,30 @@
 namespace heitech.zer0mqXt.core
 {
     public interface ISocket : IDisposable
-    { 
+    {
+         Task TryRequestAsync<TRequest, TResult>(TRequest request, Func<TResult, Task> successCallback, Func<Task> failureCallback)
```

Comment:
since this method is prefixed with _"Try"_, which usually returns a boolean indicating success/failure, would it make sense here to return a `Task<bool>` instead?
- would this be a useful convenience to the user?
- or would this be an unwanted duplication since the user can determine success/failure via the respective callbacks already?

my preference is to have this return a `Task<bool>` due to the _"Try"_ in the name, and also to be in sync with the other _"Try"_ methods in the `ISocket`

Reply:
I do not like the idea too much to mingle two different approaches (callbacks  and return values in this case) Yet I still added the boolean return value, as I have currently no better solution at hand with regards to a consistent interface but also the convenience of invoking callbacks instead of a new ResultType or such.